apps/web-giddh/src/app/shared/subscription-upgrade-button/subscription-upgrade-button.component.ts

```
@@ -0,0 +1,80 @@
+        }
+    }
+
+    /**
+     * Determines whether to show the upgrade message
+     * Checks for module restrictions and remaining count if provided
+     *
+     * @returns {boolean} True if message should be shown
+     * @memberof SubscriptionUpgradeButtonComponent
+     */
+    public shouldShowMessage(): boolean {
+        const hasRestriction = this.activeCompany?.subscription?.planDetails?.restrictedModules.hasOwnProperty(this.restrictedModule);
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)